app/controllers/posts_controller.rb

```diff
@@ -1,31 +1,29 @@
-class PostsController < ApplicationController
-  def index
-    @user = User.find(params[:user_id])
-    @posts = @user.posts
-  end
+class PostsController < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments, dependent: :destroy
```

Comment:
- [ ] Kindly use CanCanCan for Post authorization in `Post Controller`. Based on project requirements, You're to use CanCanCan for this authorization.
- You need to implement the post-deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### Requirement:

### Sample CanCanCan Authorization:
```ruby
  class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```
- Don't forget to add the delete action also as you did in comments controller